Earthfile

```diff
@@ -33,12 +37,23 @@ repos-file:
   RUN echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/ros2.list > /dev/null
 
   # Install repos file generator script requirements.
-  RUN apt-get install -y python3-rosinstall-generator ruby
+  RUN apt-get install -y python3-rosinstall-generator
+
+repos-file:
+  FROM +setup
+
+  # Copy inputs and generate a repos file
   WORKDIR /root
   COPY --dir scripts/ ./
   COPY excluded-pkgs.txt ./
   COPY spaceros-pkgs.txt ./
   COPY spaceros.repos ./
-  RUN --no-cache sh scripts/generate-repos.sh
-  RUN --no-cache ruby scripts/merge-repos.rb
-  SAVE ARTIFACT ros2.repos AS LOCAL ros2.repos
+  RUN --no-cache sh scripts/generate-repos.sh \
```

Comment:
Not blocking at all, but does `--no-cache` work the same unintuitive way in Earthly as it does in Docker, where it doesn't generate cache _hits_ but it does generate _new cache_? Might be worth figuring out a way to avoid this in future since if so it'll make people run out of disk space real fast.

Reply:
> --no-cache
Force the command to run every time; ignoring the layer cache. Any commands following the invocation of RUN --no-cache, will also ignore the cache. If --no-cache is used as an option on the RUN statement within a WITH DOCKER statement, all commands after the WITH DOCKER will also ignore the cache.

*Sighs*. Good point. Going forward, I've had success using things like mentioned here https://github.com/moby/moby/issues/1996#issuecomment-185872769 to explicitly invalidate the cache with build args, if desired. Helpful for keeping local development from blowing up but still ensures that things run if you need them to in CI.

Reply:
To me the docs were not clear on that so I' verified it locally with some sample Earthfile and indeed it creates a new cache entry even if the content of the saved artifact remains the same.